rust/pact_verifier/src/pact_broker.rs

```diff
@@ -31,24 +25,22 @@ fn as_string(json: &serde_json::Value) -> String {
     }
 }
 
-fn content_type<T>(response: &Response<T>) -> String {
+fn content_type(response: &reqwest::async::Response) -> String {
     match response.headers().get("content-type") {
         Some(value) => value.to_str().unwrap_or("text/plain").into(),
         None => s!("text/plain")
     }
 }
 
-fn json_content_type<T>(response: &Response<T>) -> bool {
-    match response.headers().get("content-type") {
-        Some(value) => {
-            match value.to_str() {
-                Err(_) => false,
-                Ok("application/json") => true,
-                Ok("application/hal+json") => true,
-                _ => false
-            }
-        },
-        None => false
+fn json_content_type(response: &reqwest::async::Response) -> bool {
+    match content_type(response).parse::<mime::Mime>() {
+        Ok(mime) => {
+            mime.type_() == mime::APPLICATION && (
+                mime.subtype() == "hal" ||
```

Comment:
I think the sub type here would be `hal+json`

Reply:
I pushed a fix. With `mime` crate the `+json` part is the suffix.